File: Python/NumComplex.py
class NumComplex:
    def __init__(self, real, imaginary):
        self.real = real
        self.imaginary = imaginary
    def get_real(self):
        return self.real
    def get_imaginary(self):
        return self.imaginary
    def add(self, other):
        newReal = self.get_real() + other.get_real()
        newImaginary = self.get_imaginary() + other.get_imaginary()
        return NumComplex(newReal, newImaginary)
    def subtract(self, other):
        newReal = self.get_real() - other.get_real()
        newImaginary = self.get_imaginary() - other.get_imaginary()
        return NumComplex(newReal, newImaginary)
    def multiply(self, other):
        newReal = self.get_real() * other.get_real()
        newImaginary = self.get_imaginary() * other.get_imaginary()
        return NumComplex(newReal, newImaginary)
    def division(self, other):
        denominator = other.get_real() * other.get_real() + other.get_imaginary() * other.get_imaginary()
        newReal = (self.get_real() * other.get_real() + self.get_imaginary() * other.get_imaginary())/denominator
        newImaginary = (self.get_real() * other.get_real() - self.get_imaginary() * other.get_imaginary())/denominator
        return NumComplex(newReal, newImaginary)
    def __str__(self) -> str: #Método que permite definir uma representação em string personalizada para nossa classe, achei lindo 
        if self.get_imaginary() >= 0:
            return f"{self.get_real()} + {self.get_imaginary()}i"
        else:
            return f"{self.get_real()} - {-self.get_imaginary()}i"

    
real1, imaginary1 = map(float, input("Digite o primeiro número complexo (parte real e parte imaginária separadas por espaço): ").split(" "))
real2, imaginary2 = map(float, input("Digite o segundo número complexo (parte real e parte imaginária separadas por espaço): ").split(" "))
num1 = NumComplex(real1, imaginary1)
num2 = NumComplex(real2, imaginary2)


sum_result = num1.add(num2)
difference_result = num1.subtract(num2)
product_result = num1.multiply(num2)



print("Resultado da Soma:", sum_result)
print("Resultado da Subtração:", difference_result)
print("Resultado da Multiplicação:", product_result)

File: Java/NumComplex.java
public class NumComplex{
    private float real;
    private float imaginary;

    public NumComplex(float other, float y){
        this.real = other;
        this.imaginary = y;
    }

    public float getReal(){
        return this.real;
    }
    public float getImaginary(){
        return this.imaginary;
    }
    public NumComplex add(NumComplex other){
        float newReal = this.getReal() + other.getReal();
        float newImaginary = this.getImaginary() + other.getImaginary();
        return new NumComplex(newReal, newImaginary);
    }

    public NumComplex subtract(NumComplex other){
        float newReal = this.getReal() - other.getReal();
        float newImaginary = this.getImaginary() - other.getImaginary();
        return new NumComplex(newReal, newImaginary);
    }
    public NumComplex multiply(NumComplex other){
        float newReal = this.getReal() * other.getReal();
        float newImaginary = this.getImaginary() * other.getImaginary();
        return new NumComplex(newReal, newImaginary);
    }
    public NumComplex division(NumComplex other){
        float denominator = other.getReal() * other.getReal() + other.getImaginary() * other.getImaginary();
        float newReal = (this.getReal() * other.getReal() + this.getImaginary() * other.getImaginary())/denominator;
        float newImaginary = (this.getReal() * other.getReal() - this.getImaginary() * other.getImaginary())/denominator;
        return new NumComplex(newReal, newImaginary);
    }

    public void printNumComplex(){
        if (this.getImaginary() > 0){
            System.out.println(this.getReal() + "+" + this.getImaginary() + "i");
        } else{
            System.out.println(this.getReal() + " " + this.getImaginary() + "i");
        }
    }
}